### PR TITLE
REQUEST FOR CODE REVIEW ON NEWSWEEK CLONE

### PR DESCRIPTION
**Pull request by Moreneecoder**
In this project, we cloned the Newsweek homepage using bootstrap CSS. We set up the linters and GitHub workflow to meet the required technical checks put in place by Microverse.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          <div class="col-3 my-auto d-none d-lg-block">
+            <p class="text-white">Wed, Jan 20, 2021</p>
+          </div>
+
+          <div class="offset-lg-1 col-lg-4 col-md-3 col-sm-4 col-4">
+            <img class="img-fluid" src="./assets/newsweek-logo1.png" alt="Newsweek logo" >
+          </div>
+
+          <div class="col-lg-4 col-md-9 col-sm-8 col-8 text-right d-flex align-items-end justify-content-end my-auto">
+            <i class="d-md-none fas fa-user mr-3" style="font-size: 1.2em; color: lightgray;"></i>
```
Comment (jubaan) [inline index.html]: - [x] Icon in tablet and mobile view also should be a link

File: index.html [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            <li class="nav-item mr-3"><a class="text-body font-weight-bold border-right border-secondary pr-3" href="nav-link">Culture</a></li>
+            <li class="nav-item mr-3"><a class="text-body font-weight-bold border-right border-secondary pr-3" href="nav-link">Newsgeek</a></li>
+            <li class="nav-item mr-3"><a class="text-body font-weight-bold border-right border-secondary pr-3" href="nav-link">Sports</a></li>
+            <li class="nav-item mr-3"><a class="text-body font-weight-bold border-right border-secondary pr-3" href="nav-link">Health</a></li>
+            <li class="nav-item mr-3"><a class="text-body font-weight-bold border-right border-secondary pr-3" href="nav-link">The Debate</a></li>
+            <li class="nav-item mr-3"><a class="text-body font-weight-bold border-right border-secondary pr-3" href="nav-link">Experts</a></li>
+            <li class="nav-item mr-3"><a class="text-body font-weight-bold" href="nav-link">Vantage</a></li>
+            <li class="nav-item mr-3"><a class="text-black-50" href="nav-link">Search</a> <i class="fas fa-search"></i></li>
```
Comment (jubaan) [inline index.html]: - [x] `<a>` anchor tag `href` attribute is used to hold ad URL or point to an `id`, it should not contain a Bootstrap class. When there is no URL or id to point at is best to use it like this within the anchor tag `href="#"`. Review all your links and make the corresponding changes.

File: style.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.body {
+  margin: 0;
+  padding: 0;
+}
```
Comment (jubaan) [inline style.css]: - [x] When using Bootstrap there is no need to reset the browser's default stylesheet as bootstrap already normalizes or resets these values.

File: index.html (hunk 2 above)
Comment (jubaan) [inline index.html]: - [ ] The search should be made from a form with a search input.

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## :wave: :wave: Hello @MoBello19 and @German-Cobian. Thanks for submitting you project!

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] Commit messages should be descriptive and meaningful. Example: 
```sh
#BAD
Update README.md

#GOOD
Update SetUp Section in README
Add author contact info to README
```
- [x] Project is not passing all linters check
- [x] Remove and add to `.gitignore` all unnecessary files from repository. `.DS_Store` in root and assets directories.
- [x] MIT license link is broken. Please create a LICENSE file and link it in the README.

**Clone**
image

**Original**
image

- [ ] **Desktop**
  - [x] elements in the Newsweek logo section doesn't have the same vertical alignment as the original
  - [x] The Newsweek logo should have the same background color as the section
  - [x] Some menu elements are taking more space than the assigned space, making them take a second row.
  - [ ] text elements in the red area don't match a similar style as the original.
  image

  - [x] Footer menu items should accommodate to use the available space and move accordingly to the screensize
  image


- [ ] **Tablet**
  - [x] Elements are not properly aligned
  image
  image

  - [x] Footer elements Logo, copyright, and social media are not aligned to equally the left.
  - [x] The inline space for the same elements is unproportioned. 
  - [x] Editions and About us paragraph are not equally aligned to the left.
  - [x] Logo, copy and social media icons should resize accordingly to the screensize
  image

- [ ] **Mobile**
  - [x] Elements layout dont match original design
  - [x] Elements are unaligned
  - [x] Suscribe link is distorted and is Unproportioned
  image
  - [ ] The footer has the same issues as the Tablet view
  image

- [x] Both header and footer logos should also be links
- [ ] There are more issues, please see inline comments.

**NOTE:** The project will appear as approved in your Microverse progress dashboard. This doesn't mean that you shouldn't work on fixing these observations, so work on them before moving to the next Milestone.




## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

**Review by ghost — COMMENTED** [review]
# Status: Invalid

**Hello team, I marked this milestone as invalid because it doesn't have a descriptive README file. You can use the Microverse readme [template](https://github.com/microverseinc/readme-template) as a guideline.**

**Submit for another review after you have implemented the above change.**

## Happy coding 🥂.

**You can reach me on slack (Rose Wanjohi).**

**Comment by ghost** [comment]
## Update

**Hello team, I wrongly invalidate your pull request because I checked the `main` branch instead of the `newsweek-clone` branch. I truly apologise for this. Please send for another review.**

File: style.css [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+.top-story-image {
+  height: 250px;
+  background-image: url("assets/latest-news-5.webp");
+}
+
+.my-turn-item a {
+  text-decoration: none;
+}
```
Comment (ghost) [inline style.css]: Avoid using custom styling where bootstrap styling is possible. Example in the above, you can use `text-decoration-none`  bootstrap classes. Please check the rest of your styling as well.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required 🛑

## Hello team 👋,

**Great job on this milestone 👏.**

## Highlights

- **You have used more bootstrap styling that custom styling.**
- **You READme is professional.**

## Improvements to note moving forward ▶️ 

- **Improve the design of the Opinion section so it looks like the original.**
- **Add the following sections in column three: Sponsored insights / Rankings 🔹 Free signup.**
- **Fix the order in which your sections display in the mobile screen size so it matches that of the original page. For example, the section at the topmost page should be Top Story followed by Featured Stories, The Debate... You started with the Featured Stories, then Sponsored insights / Rankings, Top Story...**
- **Apart from Featured Stories, Top Story and Sponsored insights / Rankings sections, all the other sections are not mobile responsive.**
- **Fix linter error checks.**
- **All images and titles should be links. You have not implemented this in a few of the sections e.g Featured stories, More Stories, Latest News and Top Story.**
- **Add a getting started section in your READme with details on how to set up your project locally.**

## Happy coding 🥂.

**You can reach me on slack (Rose Wanjohi).**

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey team, 👋

You have done an amazing job implementing the requirements for this project.

However, there are still some changes 👇 you need to implement to approve this request 🔍

### Changes Required ♻️ 

- [ ] The main Navigation menu is missing a form element with a search input. Note that it should be a form element.
- [x] Your images are missing links, almost all your images should have links.
- [x] The color of your navbar is not quite accurate. It should be red but yours in pink. Fix this
- [x] At the very bottom of your page in the `Subscribe` section, One of the items titled `NEWSLETTER` which is supposed to be towards the right is missing. 

#### Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime** or
 <a href="mailto:phillipmusiime@gmail.com"> gmail</a> &nbsp; &nbsp; [Twitter Follow](https://twitter.com/intent/follow?screen_name=Phillip_Ug) &nbsp; &nbsp; [GitHub followers](https://github.com/PhillipUg)

**Review by uryelah — APPROVED** [review]
Hi **Morenikeji Bello & German Cobian Moreno**, I'm Sarah your reviewer for this PR 👩‍🚀 

You did a great job updating your project according to the last TSE requests, the responsiveness  is working great!

Good luck with your HTML&CSS Capstones! 🚀  

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**

---------

### Optional changes(due to not being noted before by previous TSEs)

- [ ] [OPTIONAL] I see that you are using the `text-danger` and `button-danger` sometimes to get a red text and background in your clone, while using Bootstrap classes it's great it can be a problem in this case.
The shade of red the NewsWeek site uses is not the same red the Bootstrap classes have.

Please go through your clone and make sure that all the places where the shade of red differ from the original site are fixed.

### For your next projects
- [ ] Make sure your next commits follow the best committing formatting practices.
They should all be written in the imperative present mood and be capitalized:
```
// Good
Add first commit

// Bad
adding first commit
```

---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**